bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/handler/AccountHandler.java

```diff
@@ -0,0 +1,656 @@
+            logger.debug(
+                    "RestClient reported AuthenticationExceptionfrom api.ring.com when retrying refreshRegistry for the second time: {}",
+                    ex.getMessage());
+        } catch (JsonParseException ignored) {
+            logger.debug(
+                    "RestClient reported JsonParseException api.ring.com when retrying refreshRegistry for the second time: {}",
+                    ignored.getMessage());
+
+        }
+    }
+
+    /**
+     * Refresh the profile every 20 minutes
+     */
+    protected void startSessionRefresh(int refreshInterval) {
+        logger.debug("startSessionRefresh {}", refreshInterval);
+        runnableToken = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    if (restClient != null) {
+                        if (registry != null) {
+                            refreshRegistry();
+                        }
+                        // restClient.refresh_session(userProfile.getRefreshToken());
+                        Configuration config = getThing().getConfiguration();
+                        String hardwareId = (String) config.get("hardwareId");
+                        userProfile = restClient.getAuthenticatedProfile("", "", userProfile.getRefreshToken(), "",
+                                hardwareId);
+                    }
+                } catch (Exception e) {
+                    logger.debug(
+                            "AccountHandler - startSessionRefresh - Exception occurred during execution of refreshRegistry(): {}",
+                            e.getMessage(), e);
+                }
+            }
+        };
+
+        runnableEvent = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    eventTick();
+                } catch (final Exception e) {
+                    logger.debug(
+                            "AccountHandler - startSessionRefresh - Exception occurred during execution of eventTick(): {}",
+                            e.getMessage(), e);
+                }
+            }
+        };
+
+        jobTokenRefresh = scheduler.scheduleWithFixedDelay(runnableToken, 90, 600, TimeUnit.SECONDS);
+        eventRefresh = scheduler.scheduleWithFixedDelay(runnableEvent, refreshInterval, refreshInterval,
+                TimeUnit.SECONDS);
+    }
+
+    protected void stopSessionRefresh() {
+        ScheduledFuture<?> job = jobTokenRefresh;
+        if (job != null) {
+            job.cancel(true);
+        }
+        jobTokenRefresh = null;
+
+        job = eventRefresh;
+        if (job != null) {
+            job.cancel(true);
+        }
+        eventRefresh = null;
+    }
+
+    String getLocalMAC() throws Exception {
+        // get local ip from OH system settings
+        String localIP = networkAddressService.getPrimaryIpv4HostAddress();
+        if ((localIP == null) || (localIP.isEmpty())) {
```

Comment:
Maybe it's better to check for `.isBlank()` here, a string containing only whitespaces would also be invalid

Reply:
Will do. Also, changed this to IOException which seems more appropriate.